sparkmagic/sparkmagic/tests/test_configuration.py

```diff
@@ -9,7 +9,16 @@
 
 def _setup():
     conf.override_all({})
-    
+
+@patch('hdijupyterutils.configuration._load', return_value = {'required_session_configs': {'conf': {'spark.yarn.tags': "viaduct.ai/created-by=vaatu-raava"}}})
+@with_setup(_setup)
+def test_configuration_override_required(hdijupyterutils_load):
+    kpc = { 'username': 'U', 'password': 'P', 'base64_password': 'cGFzc3dvcmQ=', 'url': 'L', "auth": AUTH_BASIC }
+    overrides = { conf.kernel_python_credentials.__name__: kpc }
+    conf.override_all(overrides)
+    conf.override_required()
+    assert_equals(conf.d, {'kernel_python_credentials': {'username': 'U', 'password': 'P', 'base64_password': 'cGFzc3dvcmQ=', 'url': 'L', 'auth': 'Basic_Access'}, 'session_configs': {'conf': {'spark.yarn.tags': "viaduct.ai/created-by=vaatu-raava"}}})
```

Comment:
We should aim to contribute this back to the main `sparkmagic` repo. Can you remove the `viaduct.ai` references?

Reply:
+1 - one issue with contributing this generally, though is that we have our changes to use `default` instead of `None` for the session name